contents/security/passwordStorage.md

```diff
@@ -0,0 +1,106 @@
+* It is common for users to re-use the same password on multiple sites. If your passwords are stored as plaintext and accessed, attackers can easily re-use the same passwords on other websites (such as banking websites) and gain access to those.
+
+To prevent all of these measures, it is good to perform hashing on passwords. Unlike encryption, hashing is a <trigger for="pop:oneway">one-way function</trigger>. This means that an attacker cannot decrypt the encrypted password to retrieve the original password, thus ensuring that all of the above vulnerabilities isn't possible.
+
+<popover id="pop:oneway" title="" placement="top">
+<div slot="content">
+	A one way function is a function that is easy to compute in one way, but is close to impossible to compute the other way.
+</div>
+</popover>
+
+## What is hashing?
+
+Hashing is the act of transforming a set of data into another set of data that is difficult to reverse. As an example, suppose you wanted to multiply 2 prime numbers together, such as 17 and 11. This is easy for a computer to calculate, and it results in 187. However, suppose you were given a number that is the result of 2 prime numbers - such as 299. How do you find out which 2 prime numbers multipled together to create 299?
+
+It definitely isn't 2, because the number isn't even.  
+It isn't 3, because the digits don't sum to a multiple of 3.  
+It isn't 5, because it doesn't end in a 5 or a 0.  
+Is it 7? Well, it's hard to tell, you would have to start doing division to test.  
+ 
+There are a few shortcuts that computers and mathematicans can use to make the operation faster, but overall it is a long process that isn't efficient at all, particularly for huge numbers. Thus, it is easy to perform the function one way, but hard to reverse. 
+
+Computers use a similar forumla to manipulate the password to another random string, one that is easy to generate from the original, but would take an extremely long time to generate the original from the hash.
+
+Some examples of hashing formulas are [MD5](https://en.wikipedia.org/wiki/MD5) and [SHA256](https://en.wikipedia.org/wiki/SHA-2).
+
+### Why isn't hashing enough?
+
+Unfortunately, [rainbow tables](https://en.wikipedia.org/wiki/Rainbow_table) exist.
+
+A rainbow table is a precomputed table of hashes for some set of passwords. Basically, since the hashing function is easy to compute one-way, people build huge tables of hashes wherein the plaintext is already known, so that attempting to crack hashes becomes a simple problem of looking up the hash in the table and it's corresponding value, instead of attempting to reverse the hash. Because of this, it is very easy to crack simple hashes by simply doing a lookup.
+
+An example of a service that provides this is [Crackstation](https://crackstation.net/).
+
+Since attackcs like rainbow tables exist, passwords need another layer of security.
```

Comment:
`attackcs` typo

Also, would rainbow table be considered an "attack"?

Reply:
I think that for the purposes of this article, it should be ok to consider it an attack.